src/FileReader.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer;
+
+/**
+ * File reader that unify access to regular file and stdin-alike file.
+ *
+ * Regular file could be read multiple times with `file_get_contents`, but file provided on stdin can not.
+ * Consecutive try will provide empty content for stdin-alike file.
+ * This reader unifies access to them.
+ *
+ * @internal
+ */
+final class FileReader
+{
+    /**
+     * @var ?self
+     */
+    private static $instance;
+
+    /**
+     * @var ?string
+     */
+    private $stdinContent;
+
+    /**
+     * @return self
+     */
+    public static function createSingletone()
```

Comment:
It should be Singleton, without the `e` (https://en.wikipedia.org/wiki/Singleton_pattern).

How is that an excellent developer like you doesn't know how to properly spell such an important and fundamental design pattern? (just joking :P)

Reply:
nice catch, will fix later for sure :+1:
so much used to DIC ;P I was angry for writing this shit down, but well, stdin stream has global state :( (and in my opinion singleton === global)

Reply:
I think you are too good.

STDIN support was introduced in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/310 and from then on I've never seen a use case that wouldn't be better handled with normal command argument.

Now (a lot of versions after v0.5.0, the first tag containing it) we have better verbosity and diff to know what is going on without altering the real code.

So, if it would be up to me, I would just drop it, not because the feature is totaly useless, but because it is becoming a burden more than a valuable feature.

Regarding the implementation:

1. Singletons should have private contructor
1. I would drop Singleton at all
1. I would rely only on the public static method, after making the class final with the contructor set to private; calling `createSingleton` every time doesn't bring value to me for such a simple class

Reply:
There were actually multiple issues/questions about stdin even for already v2.
Even one with a guy having troubles using it, but we thought it was different issue. And in the end, it was possible to run stdin in very narrow configuration, like no cache and php7 linting (well... CI and tools integrations? :D)
Having in mind this and [that](https://youtrack.jetbrains.com/issue/WI-38820), I wouldn't drop the feature.

About implementation concerns. Let us go from... bottom.
> I would rely only on the public static method

nope ;) you can't test it really. That's exactly the case why I did created regular class, why i'm not using static property for data read from stdin (as it would be shared across all instances, causing tests to be problematic), why I don't hide constructor and so.
I want to have this class as "clean" as possible with one violation only - `createSingleton`, which gonna be removed as soon as we will switch to dependency injection and we will treat `FileReader` as a service. Then, the only change to `FileReader` would be to drop static `createSingleton`, not redesign of whole class (even when it's small one), which will help us in LTS branch merging